### PR TITLE
[HttpClient] Add support for NTLM authentication

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Someone mentioned NTLM auth on Twitter, so here we are, this adds support for it.
Requires curl.

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/HttpClient/HttpClientTrait.php [hunk 1]
```diff
@@ -181,6 +181,10 @@ private static function mergeDefaultOptions(array $options, array $defaultOption
                 }
             }
 
+            if ('auth_ntlm' === $name) {
+                throw new InvalidArgumentException(sprintf('Option "%s" is not supported by %s, try using CurlHttpClient instead.', __CLASS__));
```
Comment (apfelbox) [inline src/Symfony/Component/HttpClient/HttpClientTrait.php]: The option name is missing in the `sprintf()` call?

2 placeholders but only 1 value passed.

Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/HttpClientTrait.php]: good catch, could you please send a PR to fix it?

Comment (apfelbox) [reply src/Symfony/Component/HttpClient/HttpClientTrait.php]: Sure thing: https://github.com/symfony/symfony/pull/32476

**Pull request merged**